spec.bs

```diff
@@ -2871,6 +2933,217 @@ and [=map/values=] are [=moments=] at which the cool down for the origin key exp
+  |histogram|, and an [=origin=] |frameOrigin|:
+
+  1. Let |totalBuckets| be the sum of [=number of user buckets=] and [=number of platform buckets=].
+  1. [=Assert=] |histogram|'s [=list/size=] is |totalBuckets|.
+  1. Let |userHistogram| and |platformHistogram| be new [=lists=] of [=booleans=].
+  1. [=list/For each=] |i| in [=the range=] 0 to |totalBuckets|, exclusive:
+    1. If |i| &lt; |userHistogram|, then [=list/append=] |histogram|[|i|] to |userHistogram|.
+    1. Otherwise, [=list/append=] |histogram|[|i|] to |platformHistogram|.
+
+  1. Let |body| be a new [=ordered map=] of the following [=map/entries=]:
+    :   "version"
+    ::  1
+    :   "histogram"
+    ::  a new [=ordered map=] of the following [=map/entries=]:
+      :   "buckets"
+      ::  the result of [=bit packing=] with |userHistogram|
+      :   "length"
+      ::  [=number of user buckets=]
+    :   "platformHistogram"
+    ::  a new [=ordered map=] of the following [=map/entries=]:
+      :   "buckets"
+      ::  the result of [=bit packing=] with |platformHistogram|
+      :   "length"
+      ::  [=number of platform buckets=]
+  1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  «`Content-Type`: `application/cbor`»
+    :   [=request/method=]
+    ::  `POST`
+    :   [=request/body=]
+    ::  the [=byte sequence=] resulting from [[RFC8949#name-specification-of-the-cbor-e|CBOR encoding]]
+        |body|
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/origin=]
+    ::  |frameOrigin|
+    :   [=request/mode=]
+    ::  "`no-cors`"
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/credentials mode=]
+    ::  "`omit`"
+    :   [=request/redirect mode=]
+    :: "`error`"
+
+    Issue: Stop using "`no-cors`" mode where possible
+    (<a href="https://github.com/WICG/turtledove/issues/667">WICG/turtledove#667</a>).
+  1. [=Fetch=] |request| with [=fetch/useParallelQueue=] set to true.
+</div>
+
+<div algorithm>
+  To <dfn>send real time reports</dfn> given a [=real time reporting contributions map=]
+  |contributionsMap| and an [=origin=] |frameOrigin|:
+
+  1. [=map/For each=] |origin| → |contributions| of |contributionsMap|:
+    1. Let |maybeBucket| be the result of [=sampling real time contributions=] with |contributions|.
+    1. Let |histogram| be the result of [=applying RAPPOR noise=] with |maybeBucket|.
+    1. Let |reportUrl| be a new [=URL=] with the following [=struct/items=]:
+      :   [=url/scheme=]
+      ::  |origin|'s [=origin/scheme=]
+      :   [=url/host=]
+      ::  |origin|'s [=origin/host=]
+      :   [=url/port=]
+      ::  |origin|'s [=origin/port=]
+      :   [=url/path=]
+      ::  « ".well-known", "interest-group", "real-time-report" »
+    1. [=Send a real time report=] with |reportUrl|, |histogram| and |frameOrigin|.
+
+      Issue: TODO: Spec rate limiting.

```

Comment:
that gets us into rate limit error on github quite often, so we changed back to the less fancy way.

Reply:
That doesn't sound great - have you filed a bug somewhere? Maybe bikeshed repo?

Reply:
From [bikeshed spec](https://speced.github.io/bikeshed/#cli-spec): "If you’re using [Inline GitHub Issues](https://speced.github.io/bikeshed/#metadata-inline-github-issues) heavily, you might run into the GitHub rate limit. You can generate an OAuth token at https://github.com/settings/tokens and then pass it to Bikeshed via this flag to raise your limit considerably". So it's expected, but just not sure how "heavily" is defined, since only one inline issue caused rate limiting errors a few times (not very often, but still...).

I'd assume it uses Github's [REST API for search](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-issues-and-pull-requests), which is [rate limited](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#rate-limit) to 10 requests per minute.